app/Profile.php

```diff
@@ -291,10 +294,16 @@ public function updateDatum($section, $request)
 
       }
 
-        // update overall profile visibility
-        if ($section == 'information' && $request->hasAny(['public', 'full_name'])) {
+        // update overall profile record
+        if ($section == 'information' && $request->hasAny(['public', 'type', 'full_name'])) {
             $this->update([
                 'public' => $request->input('public') ?? $this->public,
+                'type' => match ($request->input('type')) {
```

Comment:
To update the profile type, could we use the value from the request directly if we assign the enum value for each option directly in the view instead of matching the values?

Reply:
Yes, that would simplify things, but using a separate value in the request is better for security.